src/lua/thread.rs

```diff
@@ -152,10 +152,24 @@ pub fn init() {
         info!("Skipping config search");
     }
 
+    // Only ready after loading libs
+    *RUNNING.write().expect(ERR_LOCK_RUNNING) = true;
+    debug!("Entering main loop...");
+    let _lua_handle = thread::Builder::new()
+        .name("Lua thread".to_string())
+        .spawn(move || { main_loop(receiver, &mut lua) });
+    // Immediately update all the values that the init file set
+    send(LuaQuery::UpdateRegistryFromCache)
+        .expect("Could not update registry from cache");
+}
+
+pub fn on_compositor_ready() {
+    info!("Running lua on_init()");
     // Call the special init hook function that we read from the init file
-    if let Err(error) = lua.execute::<()>(INIT_LUA_FUNC) {
-        error!("Lua init callback returned an error: {:?}", error);
-    }
+    send(LuaQuery::Execute(INIT_LUA_FUNC.to_owned()))
+        .err()
+        .map(|error| { error!("Lua init callback returned an error: {:?}", error); error });
+    
```

Comment:
When I pulled it down in my editor, it showed there whitespace on that blank line (165). It's not a big deal, I can always strip it if I notice any after I merge. I really should set up an auto-formatter for the project soon.